eppo_core/src/event_ingestion/event_delivery.rs

```diff
@@ -27,9 +27,9 @@ pub(super) enum EventDeliveryError {
     NonRetriableError(reqwest::Error),
 }
 
-#[derive(Debug, Serialize, Deserialize)]
-struct IngestionRequestBody {
-    eppo_events: Vec<Event>,
+#[derive(Debug, Serialize)]
+struct IngestionRequestBody<'a> {
+    eppo_events: Vec<&'a Event>,
```

Comment:
This is going half-way for avoiding copy. It does not copy events but it copies the structure of the vector (~80kb with max batch).

Ideally, we want it to take a slice:
```suggestion
    eppo_events: &'a [Event],
```


This is likely complicated by `QueuedEvent` being different from a normal `Event` but I think we can batch `Event`s directly. (My understanding is that `QueuedEvent` doesn't serve any purpose as we're just discarding all the extra fields. `BatchedMessage` was designed to be used with `Event` directly. Retried events and attempts should be scoped to the retry component.)